### PR TITLE
Print <yyyy>-<index> info for --version option

### PR DESCRIPTION
**Pull request by Qining**

### CODE REVIEW

**Comment by Qining** [comment]
@AWoloszyn @antiagainst @dekimir @dneto0 PTAL

Just copied the 'deduceSoftwareVersion()' function from Spirv-tools repo and added the CHANGES file for shaderc.

Choose 'coping instead of importing' because we also need to get the <yyyy>-<index> string for shaderc, and importing a function from spirv-tools/utils/update-build-versions.py to do the job for shaderc looks really weird, I think.

File: utils/update_build_version.py [hunk 1]
```diff
@@ -71,21 +89,32 @@ def describe(dir):
             return 'unknown hash, ' + datetime.date.today().isoformat()
 
 
+def getVersionString(project, dir):
```
Comment (antiagainst) [inline utils/update_build_version.py]: Should be named as `get_version_string`.

File: CHANGES [hunk 2]
```diff
@@ -0,0 +1,16 @@
+Revision history for Shaderc
+
+v2016.0-dev 2016-05-19
+
+ - Adds v<year>.<index> versioning, with "-dev" indicating
+   work in progress.  The intent is to more easly report
+   and summarize functionality when SPIRV-Tools is incorporated
```
Comment (dneto0) [inline CHANGES]: You mean Shaderc.

File: CHANGES [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+   in downstream projects.
+
+ - Summary of functionality (See the README.md for more):
+   - Compiles GLSL to SPIR-V binary through both command line and API calls
+   - Supports compiling GLSL to SPIR-V assembly code
+   - Supports compiling SPIR-V assembly code to SPIR-V binary
+   - Supports OpenGL, OpenGL Compatible and Vulkan semantics
+   - Supports user-definined macros
```
Comment (dneto0) [inline CHANGES]: "user-defined"

File: CHANGES [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+   - Supports dependency information dumpping
```
Comment (dneto0) [inline CHANGES]: "dumping"

File: CHANGES [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+ - Adds v<year>.<index> versioning, with "-dev" indicating
+   work in progress.  The intent is to more easly report
+   and summarize functionality when SPIRV-Tools is incorporated
+   in downstream projects.
+
+ - Summary of functionality (See the README.md for more):
+   - Compiles GLSL to SPIR-V binary through both command line and API calls
+   - Supports compiling GLSL to SPIR-V assembly code
+   - Supports compiling SPIR-V assembly code to SPIR-V binary
+   - Supports OpenGL, OpenGL Compatible and Vulkan semantics
```
Comment (dneto0) [inline CHANGES]: I'd also say "Supports #include"

File: utils/update_build_version.py (hunk 1 above)
Comment (dneto0) [inline utils/update_build_version.py]: Hm.  I see the style guide says this.
In that case we should change deduceSoftwareVersion should be changed too.
(And do the same in SPIRV-Tools in another CL).

**Comment by dneto0** [comment]
To be 100% clear, I wrote the deduceVersionString code in SPIRV-Tools, and it's owned by Google.

-1 for the errors in the CHANGES file.

File: CHANGES [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+
+v2016.0-dev 2016-05-19
+
+ - Adds v<year>.<index> versioning, with "-dev" indicating
+   work in progress.  The intent is to more easly report
+   and summarize functionality when SPIRV-Tools is incorporated
+   in downstream projects.
+
+ - Summary of functionality (See the README.md for more):
+   - Compiles GLSL to SPIR-V binary through both command line and API calls
```
Comment (antiagainst) [inline CHANGES]: We can compile GLSL to multiple targets, not only binary.

Comment (antiagainst) [inline CHANGES]: To be specific, I mean binary, c-init-list, etc. But it seems that's too trivial to mention. I'm OK to ignore.

File: CHANGES [hunk 7]
```diff
@@ -0,0 +1,17 @@
+Revision history for Shaderc
+
+v2016.0-dev 2016-05-19
+
+ - Adds v<year>.<index> versioning, with "-dev" indicating
+   work in progress.  The intent is to more easly report
+   and summarize functionality when Shaderc is incorporated
```
Comment (antiagainst) [inline CHANGES]: summarize and report.

File: CHANGES [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+   in downstream projects.
```
Comment (antiagainst) [inline CHANGES]: for incorporating into downstream projects.

File: CHANGES [hunk 9]
```diff
@@ -0,0 +1,17 @@
+Revision history for Shaderc
+
+v2016.0-dev 2016-05-19
+
+ - Adds v<year>.<index> versioning, with "-dev" indicating
```
Comment (antiagainst) [inline CHANGES]: with an optional "-dev" suffix to indiate...

File: CHANGES [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+   work in progress.  The intent is to more easly report
+   and summarize functionality when Shaderc is incorporated
+   in downstream projects.
+
+ - Summary of functionality (See the README.md for more):
```
Comment (antiagainst) [inline CHANGES]: functionalities

File: CHANGES [hunk 11]
```diff
@@ -0,0 +1,17 @@
...
+   - Compiles GLSL to SPIR-V binary through both command line and API calls
```
Comment (antiagainst) [inline CHANGES]: `Supports compiling ...` to be consistent.

**Comment by antiagainst** [comment]
+1.

**Comment by dneto0** [comment]
+2

**Comment by dneto0** [comment]
Rebased, squashed, and pushed into master as  bc1b67757

**Pull request closed without merging**